### DISCUSSION TITLE
2 pages of buttons problem

### DISCUSSION THREAD

**Discussion by neovic1** [Bodmer/TFT_eSPI#1906] [Q&A - Touch]
Hello everyone. I haven't seen a solution for this anywhere. I have created a simple main screen with 2 buttons. I want each one of them to go to another screen/page and I suceeded in that but the problem is that the first buttons always remain there (hidden but you can still touch their coordinates and make them appear) even though I clear the screen with the tft.fillscreen method. If I create new buttons on the second and third screens that are at the same position of the ones at the main screen, they overlap. Why? Anyone knows how to solve this? It seems that I need a different method to TRULY clear the screen and all its previous components.

Thanks in advance.

**Comment by Bodmer** [comment]
The simplest approach is to disable the touch handling for those buttons when the screen is changed.

**Reply by GeorgeIoak** [reply]
If I understand you correctly you're recommending that you keep track of which page you're on and which buttons are on each page. Then in this routine if the button is on a different page you just skip and don't preform a ->press().
```
for (uint8_t b = 0; b < buttonCount; b++)
    {
      if (pressed)
      {
        ledcWrite(0, ledBrightness);
        //Serial.printf("In touch loop and brightness changed%u\n", ledcRead(0));
        holdingRegs[DISP_ONOFF] = 255;  // Reset Backlight Timer
        if (btn[b]->contains(t_x, t_y))
        {
          btn[b]->press(true);
          btn[b]->longPressAction(b, 1000, 5); // btn#, long press time, long press increment
        }
      }
      else
      {
        btn[b]->press(false);
        btn[b]->releaseAction();
      }
    }
```

I did run across this interesting library that extended your code, [DisplayMenu](https://github.com/guttih/DisplayMenu.) I like how he added a ["page" to the class](https://github.com/guttih/DisplayMenu/blob/main/src/DisplayButton.cpp#L15) but I'm already using your TFT_eButton and didn't see an easy way to integrate his code.

**Reply by magmaonfire** [reply]
> The simplest approach is to disable the touch handling for those buttons when the screen is changed.

How would we accurately be able to do this? I currently have 15 buttons defined throughout three functions (for the different pages), and I have set the main menu to be the first screen in my void setup(). Then, I put the getTouch and press functions in my void loop(), followed by multiple if statements to detect when a certain button is pressed to go to another page. My problem is as soon as I set it to that next page in the if statement that is in the loop function, it still recognizes the buttons from the previous page, as stated above.

**Reply by Bodmer** [reply]
Create a variable that records the current page, then use if statements in the loop to decide which buttons are checked. 
 if (page == 1) {
    ... check page 1 buttons ...
}

**Reply by magmaonfire** [reply]
That worked! I got it to no longer bleed through on the pages! Thanks for the help!
